airflow/models/dataset_dag_ref.py

```diff
@@ -0,0 +1,60 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from sqlalchemy import Column, ForeignKeyConstraint, Integer, String
+from sqlalchemy.orm import relationship
+
+from airflow.models.base import ID_LEN, Base
+from airflow.utils import timezone
+from airflow.utils.sqlalchemy import UtcDateTime
+
+
+class DatasetDagRef(Base):
+    """References from a DAG to an upstream dataset."""
+
+    dataset_id = Column(Integer, primary_key=True, nullable=False)
+    dag_id = Column(String(ID_LEN), primary_key=True, nullable=False)
+    created_at = Column(UtcDateTime, default=timezone.utcnow, nullable=False)
+    updated_at = Column(UtcDateTime, default=timezone.utcnow, onupdate=timezone.utcnow, nullable=False)
+
+    dataset = relationship('Dataset')
+
+    __tablename__ = "dataset_dag_ref"
+    __table_args__ = (
```

Comment:
```suggestion
    __table_args__ = (
        PrimaryKeyConstraint(
            "dataset_id", "dag_id", 
            name="datasetdagref_pkey",
            mssql_clustered=True
            ),
```
This will give the primary key a name.

Reply:
done